### ISSUE TITLE
Revit: updating geometry doesn't always work & creates duplicates

### ISSUE THREAD

**Issue by clairekuang** [specklesystems/speckle-sharp#1576]
## What package are you referring to?

Revit connector 

## Describe the bug

Reported in the forum: https://speckle.community/t/civil3d-alignment-revit/3508
Also, observed while receiving the standard geometry file from rhino

Possible reaseon: check that displayvalues or created curve arrays are using the parent obj application id when checking for updates

## To Reproduce

receive the standard geometry file from rhino test stream multiple times in revit with setting set to `update`

## Expected behavior

No duplicates are created

#### Optional: Affected Projects
 potentially check for missing application ids from rhino, civil3d, autocad etc

**Comment by Ibrahim5aad** [comment]
Hey @clairekuang,

I'm able to replicate this while getting an `Alignment` that was updated in Civil3D.  

I think the bug lies in this `GetExistingElementByApplicationId` method, because an alignment isn't simply one Revit element.

![Screenshot 2022-09-13 121019](https://user-images.githubusercontent.com/50090593/189875737-aa5d1598-616d-427e-a289-fd9976a9e6ca.png)

Is there a reason (another use case maybe?) that this function returns only the first element of the `CreatedIds`?

![Screenshot 2022-09-13 121142](https://user-images.githubusercontent.com/50090593/189876542-89402bd9-37c9-45e3-b1da-33165c16ac0f.png)

An `Alignment` has numerous underlying model lines and curves, so by deleting only the first curve while receiving some updates from Speckle will create duplicates for sure.

**Comment by clairekuang** [comment]
Yup this is exactly the issue! There isn't a known use case for only retrieving the first created obj iirc, it's just a bug - will get around to fixing this soon :)
